### PR TITLE
reserved prefixes for names

### LINKED ISSUE #769: reserved prefixes for names
NeXus will develop and publish a list of reserved prefixes and avoid standardizing any name (specifically, we are thinking of attribute names) that starts with a reserved prefix.  To reserve a prefix, post a GitHub issue to the definitions repository.

Should this be generalized to prefix the names of groups, fields, and links, as well?

_Originally posted by @prjemian in https://github.com/nexusformat/definitions/issues/757#issuecomment-619031547_

### LINKED ISSUE #781: Request to reserve SILX_ name prefix
Hi,

Following #757 and according to #769 (related to #770), we'd like to reserve the name prefix `SILX_` for our applications.

attn @vasole, @kif

### LINKED ISSUE #831: Vote on proposal for NXdata axes ...
Per [NIAC2014](https://www.nexusformat.org/NIAC2014_Meeting#Minutes.html) approved vote, 

> Vote on proposal for NXdata axes with indices attribute only required when required to resolve ambiguity. All in favour. Vote on proposal with indices required as in [2014_axes_and_uncertainties](https://www.nexusformat.org/2014_axes_and_uncertainties.html).

### PR DESCRIPTION
**Pull request by prjemian**
fixes #769 and fixes #781 and fixes #831

### CODE REVIEW

**Comment by prjemian** [comment]
wait for merge pending outcome of [NIAC2020 vote](https://github.com/nexusformat/NIAC/issues/49#issuecomment-707383140)

File: manual/source/datarules.rst [hunk 1]
```diff
@@ -128,6 +128,38 @@ cryostat/furnace/whatever. The field ``temperature_set`` will hold
 the target value for the temperature as set by the 
 experiment control software. 
 
+.. index:: ! reserved prefixes
+
+.. _reserved_prefixes:
+
+.. rubric:: Reserved field name prefixes
+
+When naming a field (or dataset), NeXus has reserved certain prefixes to
```
Comment (PeterC-DLS) [inline manual/source/datarules.rst:137]: Remove "(or dataset)" to make terminolgy consistent

File: manual/source/datarules.rst [hunk 2]
```diff
@@ -128,6 +128,38 @@ cryostat/furnace/whatever. The field ``temperature_set`` will hold
...
+    reserved prefixes; PDBX_
+    reserved prefixes; SAS_
+    reserved prefixes; SILX_
+
+==========  ==========  ==========================================  ==============
+prefix 	    use 	    meaning 	                                URL
+==========  ==========  ==========================================  ==============
+``IDF_``    attributes 	reserved for use by ISIS Muon Facility 	    https://www.isis.stfc.ac.uk
+``NDAttr``  attributes 	reserved for use by EPICS area detector     https://github.com/areaDetector
+``NX``      NXDL class 	For the class names used with NeXus groups 	https://www.nexusformat.org
```
Comment (PeterC-DLS) [inline manual/source/datarules.rst:156]: consistent uncapitalization of "For"

**Review by PeterC-DLS — CHANGES_REQUESTED** [review]

**Review by yayahjb — APPROVED** [review]

**Comment by yayahjb** [comment]
A good step forward

**Review by vasole — APPROVED** [review]
Thanks

**Review by PeterC-DLS — APPROVED** [review]

**Review by benajamin — APPROVED** [review]

**Pull request merged**